src/librustc_infer/traits/error_reporting/suggestions.rs

```diff
@@ -1430,7 +1430,7 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
     crate fn note_obligation_cause_code<T>(
         &self,
         err: &mut DiagnosticBuilder<'_>,
-        predicate: &T,
+        predicate: Option<&T>,
         cause_code: &ObligationCauseCode<'tcx>,
         obligated_types: &mut Vec<&ty::TyS<'tcx>>,
```

Comment:
This is a weird name *and* type. I assume this was meant to be `&mut Vec<Ty<'tcx>>`? How is this not tripping the lint against types like `TyS`?

Reply:
cc @oli-obk @flip1995

Reply:
This should indeed trip the internal lint. I want to revisit this lint anyway, to improve its suggestions. I'll look into this then.